File: euler/04/4.scala
import scala.util.Random

def reverse(x:Int):Int={
	var xcpy = x 
	var y=0
	while(xcpy!=0){
		y *= 10
		y += xcpy%10
		xcpy = (xcpy-(xcpy%10))/10
	}
	return y
}

/*
val gen = new Random()

for(i<-0 until 5){
	val x = gen.nextInt(200)
	println(x + " " + reverse(x))
}
*/

def isPalindrome(x:Int): Boolean={
	return x==reverse(x)
}
	
val upper = 1000
var largest = 0


for(i<-0 until upper){
	for(j<-0 until i+1){
		var prod = i*j		
		//println(prod)
		if(isPalindrome(prod) && prod>largest){
			largest = prod
		}
	}
}

println("The largest product is " + largest)

File: euler/04/4.cpp
#include <cstdlib>
#include <cstdio>

int palindrome(int x){
	int y=0;
	while(x != 0){
		y = y*10 + x%10;
		x = (x - x%10)/10;
	}
	return y;
}

bool isPalindrome(int x){
	return x == palindrome(x);
}

int main(){
	const int upper = 1000;
	int largest = 0;


	for(int i=0; i<upper; i++){
		for(int j=0; j<=i; j++){
			if(isPalindrome(i*j) && i*j>largest){
				largest = i*j;
			}
		}
	}
	printf("The largest palindrome is %d\n",largest);
}
